test/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs

```diff
@@ -892,5 +893,129 @@ public void InstallCommand_DependencyResolutionFailure()
                 Util.DeleteDirectory(source);
             }
         }
+
+        // Tests that when credential is saved in the config file, it will be passed 
+        // correctly to both the index.json endpoint and registration endpoint, even
+        // though one uri does not start with the other uri.
+        [Fact]
+        public void InstallCommand_AuthenticatedV3WithCredentialSavedInConfig()
+        {
+            var nugetexe = Util.GetNuGetExePath();
+            var randomTestFolder = TestFilesystemUtility.CreateRandomTestFolder();
+            bool credentialsPassedToRegistrationEndPoint = false;
+
+            try
+            {
+                // Server setup                
+                using (var serverV3 = new MockServer())
+                {
+                    var registrationEndPoint = serverV3.Uri + "w";
+                    var indexJson = JObject.Parse(@"{
+                  ""version"": ""3.2.0"",
+                  ""resources"": [ {""@id"": """ + registrationEndPoint + @""", ""@type"": ""RegistrationsBaseUrl/3.0.0-beta""} ],
+                ""@context"": {
+                ""@vocab"": ""http://schema.nuget.org/services#"",
+                ""comment"": ""http://www.w3.org/2000/01/rdf-schema#comment""
+                    }}");
+
+                    serverV3.Get.Add("/a/b/c/index.json", r =>
+                    {
+                        var h = r.Headers["Authorization"];
+                        var credential = String.IsNullOrEmpty(h) ? 
+                            null :
+                            System.Text.Encoding.Default.GetString(Convert.FromBase64String(h.Substring(6)));
+
+                        if (StringComparer.OrdinalIgnoreCase.Equals("test_user:test_password", credential))
+                        {
+                            return new Action<HttpListenerResponse>(response =>
+                            {
+                                response.StatusCode = (int)HttpStatusCode.OK;
+                                response.ContentType = "text/javascript";
```

Comment:
The other test is wrong. We should either remove the content type or use the correct one

Sent from my Windows Phone
________________________________
From: feiling<mailto:notifications@github.com>
Sent: ‎8/‎31/‎2015 7:09 PM
To: NuGet/NuGet.PackageManagement<mailto:NuGet.PackageManagement@noreply.github.com>
Cc: Yishai Galatzer<mailto:yigalatz@microsoft.com>
Subject: Re: [NuGet.PackageManagement] Unit test for the fix of NuGet/Home#1300. (#90)


In test/NuGet.CommandLine.Test/NuGetInstallCommandTest.cs<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.PackageManagement%2fpull%2f90%23discussion_r38379943&data=01%7c01%7cyigalatz%40microsoft.com%7cb138ae0e8b5541d37e6308d2b2725703%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=TMbXacoLT4fATAnuTEkTgEERcEKC0PrWqWHxITMygvs%3d>:

> +                ""comment"": ""http://www.w3.org/2000/01/rdf-schema#comment""
> +                    }}");
> +
> +                    serverV3.Get.Add("/a/b/c/index.json", r =>
> +                    {
> +                        var h = r.Headers["Authorization"];
> +                        var credential = String.IsNullOrEmpty(h) ?
> +                            null :
> +                            System.Text.Encoding.Default.GetString(Convert.FromBase64String(h.Substring(6)));
> +
> +                        if (StringComparer.OrdinalIgnoreCase.Equals("test_user:test_password", credential))
> +                        {
> +                            return new Action<HttpListenerResponse>(response =>
> +                            {
> +                                response.StatusCode = (int)HttpStatusCode.OK;
> +                                response.ContentType = "text/javascript";


This was copied from another test. In fact the content type is never checked by our client.

—
Reply to this email directly or view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fNuGet%2fNuGet.PackageManagement%2fpull%2f90%2ffiles%23r38379943&data=01%7c01%7cyigalatz%40microsoft.com%7cb138ae0e8b5541d37e6308d2b2725703%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=YqfYwdkfMFuou2WV%2fxEybzwOsmjotgBewWvc8VLUo%2bw%3d>.